Update to latest from upstream 5.6 (02-Oct-16)

Herrie82 opened the pull request.

carewolf committed:
Disable GPU when no shared OpenGL context is available

In case an OpenGL context can not be created we can try running in
software compositing mode. This should have a much better chance of
working as a fallback.

Change-Id: Icaab24dde8f4b6c8ea4023cf36b0bb679143166f
Reviewed-by: Michael Brüning <michael.bruning@qt.io>

Commit:
Doc: Change instances of 'OS X' to 'macOS'

As of version 10.12 (Sierra), the name of Apple's desktop operating
system will be macOS. Change all occurrences where the Mac platform
is discussed to use the macro \macos (defined in the documentation
configuration in qtbase), except in link targets where QDoc does
not expand macros.

Change-Id: Icad4e7817d93b9e7b0e1a9a13a8618b3341ab2ba
Reviewed-by: Leena Miettinen <riitta-leena.miettinen@qt.io>

alcroito committed:
Fix IME window placement relative to QWebEngineView

The position of the IME window is computed using the widget input item
transform. When a regular QWidget gets a focusIn event, the input item
transform is recomputed inside the
QWidgetPrivate::updateWidgetTransform method. This did not happen
for the QWebEngineView, because the focus event is handled internally
and not passed down to QWidget::event.
Fix consists in calling updateWidgetTransform manually whenever the
view receives focus.
The other cases when updateWidgetTransform should be called (namely
resize and move events) are handled properly by delegating to
QWidget::event.

Task-number: QTBUG-55634
Change-Id: Ic93662929e169d860f8ca567f1955da4dc45f9fe
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>

alcroito committed:
Fix select tag interaction when the web view is inside a modal dialog

Previously when a QWebEngineView was inside a modal QDialog, trying to
click on a select tag option did not properly select the option. It
either focused the new option without closing the popup, or didn't
focus it at all.

Fix consists in making sure the newly created popup QWindow and
RenderWidgetHostViewQtDelegateWidget are marked as children of the
QWebEngineView, so that they are considered part of the current modal
session by the OS, thus allowing user interaction with them.

Because the ownership of the delegate widget should still be retained
by its respective RenderWidgetHostViewQt instance, the QObject parent
of the delegate is unset before the parent is destroyed.

Also to make it work on macOS, the window attribute has to be set
to Qt::Tool instead of Qt::ToolTip.

Change-Id: I56d6f446254a624428a0c661ac3c49eb409c931e
Task-number: QTBUG-54836
Reviewed-by: Qt CI Bot <qt_ci_bot@qt-project.org>
Reviewed-by: Michael Brüning <michael.bruning@qt.io>

Commit:
Doc: Fix initialize() documentation

We do not share context between "processes", but between
"threads"

Change-Id: I20f558f913c1f19fc469ade6faab45762f42d528
Reviewed-by: Kai Koehne <kai.koehne@qt.io>

alcroito committed:
Fix flaky tst_QQuickWebEngineView::loadProgress test

Change-Id: I102bda62bfaa16a8c9ff4c4929e6b5a055de3e25
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>

Tofee reviewed (approved):
Nothing urgent here, we can merge it later on with other real changes.

kkoehne committed:
Doc: Clarify that font size is in pixels

Task-number: QTBUG-56645
Change-Id: I16ca76d3a26d0142846b068cc1ba52bee71cf106
Reviewed-by: Leena Miettinen <riitta-leena.miettinen@qt.io>

AlgorithMan-de committed:
Allow Pepper Flash to access Media Devices

The Q_UNREACHABLE caused WebEngine to block Pepper Flash from
accessing Media Devices such as the WebCam (this rendered
http://testwebcam.com unusable for example)

Task-number: QTBUG-55017
Change-Id: I7979c9a5690173e86310f7448c78949371fbc909
Reviewed-by: Michael Brüning <michael.bruning@qt.io>

jobor committed:
Fix leakage of FilePickerControl objects in QWebEnginePage

Delete the FilePickerControl object after we're done with it, analog to
what the QtQuick UIDelegatesManager does.

Change-Id: Id5ef6666c8536b9e5c6877cd522b4c20adae37e5
Reviewed-by: Michal Klocek <michal.klocek@qt.io>

AlgorithMan-de committed:
Pepper flash search-path for package adobe-flashplugin

Since version 54, the chromium distribution package does not contain
the pepper flash plugin anymore. Because of this, the plugin cannot
be installed using the Ubuntu package "pepperflashplugin-nonfree"
anymore (because the package pulled the plugin from the chromium
distribution).
The plugin is also contained in the package adobe-flashplugin, which
installs the file to /usr/lib/adobe-flashplugin/libpepflashplayer.so.

[ChangeLog] Pepper flash is now also searched in /usr/lib/adobe-flashplugin/

Change-Id: I7c135a2eac1628b0d338bbfed07695c9578458e6
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>

mibrunin committed:
Copy web channel from QWebEnginePage when adopting WebContentsAdapter

It used to be just ignored.

Task-number: QTBUG-56643
Change-Id: Ife7ac05fce7e93a6eba1ac6e6db1095574c83e42
Reviewed-by: Viktor Engelmann <viktor.engelmann@qt.io>
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>

ossilator committed:
remove dependencies from sync.profile

the CI obtains them from the qt5 super repo nowadays.

Change-Id: If41f7c5306ca739fd3800cec3f680fa72df2a871
Reviewed-by: Jędrzej Nowacki <jedrzej.nowacki@qt.io>

AndyShawQt committed:
Check if the .git directory exists before invoking git

If .git does not exist then this is not from a git build and therefore
there is no reason to invoke git (which might not even exist on the
system anyway). This prevents an error appearing due to trying to invoke
git on Windows when it does not exist.

Change-Id: I8c0b5b237cfdaffdbb33efdd16cf20cd1560f1a1
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>

stampho committed:
Fix IPC message logging

Register Qt IPC messages for logging. The usage of
content::RegisterIPCLogger function adds the
content_common.content_ipc_logging.o to the link dependency. Thus it
will register the Chromium Content IPC messages too.

Task-number: QTBUG-57224
Change-Id: I2c45691feb22a34f6074940cc35b8a4ba7804370
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>

Herrie82 commented:
New tested PR without conflicts: https://github.com/webOS-ports/qtwebengine/pull/42
